src/Search.php

```diff
@@ -4573,8 +4573,38 @@ public static function addDefaultWhere($itemtype)
                     break;
                 }
 
+                // Build base condition using entity restrictions
+                $condition = "(";
+                $condition .= getEntitiesRestrictRequest(
+                    '',
+                    'glpi_tickets_items_id_' . self::computeComplexJoinID([
+                        'condition' => 'AND REFTABLE.`itemtype` = "' . Ticket::getType() . '"'
+                    ]),
+                    'entities_id',
+                    ''
+                );
+                $condition .= " OR ";
+                $condition .= getEntitiesRestrictRequest(
+                    '',
+                    'glpi_changes_items_id_' . self::computeComplexJoinID([
+                        'condition' => 'AND REFTABLE.`itemtype` = "' . Change::getType() . '"'
+                    ]),
+                    'entities_id',
+                    ''
+                );
+                $condition .= " OR ";
+                $condition .= getEntitiesRestrictRequest(
+                    '',
+                    'glpi_problems_items_id_' . self::computeComplexJoinID([
+                        'condition' => 'AND REFTABLE.`itemtype` = "' . Problem::getType() . '"'
+                    ]),
+                    'entities_id',
+                    ''
+                );
+                $condition .= ") ";
```

Comment:
Please use a loop based the hardcoded `['Ticket', 'Change', 'Problem']` array and add a TODO comment to remind us that we should use the `$CFG_GLPI['itil_types']` when this will be merged in the `main` branch.